### PR TITLE
integration specs for views

### PR DESCRIPTION
**Pull request by mumbereh**
### **This milestone implement the following**

- Integration specs
- Use Capybara to write integration tests for each view in your project
- I can see the username of all other users.
- I can see the profile picture for each user.
- I can see the number of posts each user has written.
- When I click on a user, I am redirected to that user's show page.
- user show page:
- I can see the user's profile picture.
- I can see the user's username.
- I can see the number of posts the user has written.
- I can see the user's bio.
- I can see the user's first 3 posts.
- I can see a button that lets me view all of a user's posts.
- When I click a user's post, it redirects me to that post's show page.
- When I click to see all posts, it redirects me to the user's post's index page.
- User post index page:
- I can see the user's profile picture.
- I can see the user's username.
- I can see the number of posts the user has written.
- I can see a post's title.
- I can see some of the post's body.
- I can see the first comments on a post.
- I can see how many comments a post has.
- I can see how many likes a post has.
- I can see a section for pagination if there are more posts than fit on the view.
- When I click on a post, it redirects me to that post's show page.
- Post show page:
- I can see the post's title.
- I can see who wrote the post.
- I can see how many comments it has.
- I can see how many likes it has.
- I can see the post body.
- I can see the username of each commentor.
- I can see the comment each commentor left.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,47 +1,28 @@
 class PostsController < ApplicationController
-  before_action :set_user, only: %i[index show new create]
-
+  layout 'standard'
   def index
-    @posts = @user.posts
+    @user = User.find(params[:user_id])
+    @posts = Post.where(author_id: params[:user_id]).order(id: :asc)
+    @posts = @posts.paginate(page: params[:page], per_page: 5)
```
Comment (Gambit142) [inline app/controllers/posts_controller.rb]: - [ ] Kindly solve the **_N+1 problems_** when fetching all posts and their comments for a user on the `Posts#Index` page. Solving **_N+1 problems_** in Ruby on Rails is crucial to optimize database queries, preventing excessive and inefficient database requests, thus improving application performance and responsiveness. Kindly implement this suggestion and attach the screenshots of the logs when fetching data on the `Posts#Index` page before and after the fix. 😎 

You can read this [article](https://guides.rubyonrails.org/active_record_querying.html#n-1-queries-problem:~:text=13.1%20N%20%2B%201%20Queries%20Problem) to learn how to solve N+1 problems. 👍 💻

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @mumbereh @ngalamac 👋, 

Good job so far!

### Highlights
- Setup linters correctly ✔️ 
- Wrote integration tests for `Users` controllers and view files ✔️ 
- Wrote integration tests for `Posts` controllers and view files ✔️
- All test cases pass ✔️ 

You still need to work on some issues to go to the next project, but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

You can also consider the following:

N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -1,5 +1,6 @@
 class PostsController < ApplicationController
   layout 'standard'
+
   def index
     @user = User.find(params[:user_id])
     @posts = Post.where(author_id: params[:user_id]).order(id: :asc)
```
Comment (Lucas-Masaba) [inline app/controllers/posts_controller.rb]: - [ ] As mentioned by the previous reviewer, you should solve N+1 problems here. You should use the includes method for example:
```ruby
@posts = @user.posts.includes(:comments)
```

**Review by Lucas-Masaba — CHANGES_REQUESTED** [review]
# Hello @mumbereh  👋 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
almost-there

 ## Highlights 🍾 
- Linter checks passing :heavy_check_mark: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions 💭 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏 💻 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by 1GORDON — APPROVED** [review]
# Status: APPROVED 🟢 

Hi @mumbereh, AND @ngalamac ✋

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! :tada: :clap: :clap: :100:

## Highlights!
- Implemented requested changes ✔️

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**